### PR TITLE
rpcn: add cookie impl to tickets

### PR DESCRIPTION
**Pull request by RipleyTom**
Adds support for cookies to be included in the ticket(not 1:1 with ps3 but no information could be found about cookie encoding in the ticket).

Merge needs to be synced with rpcn update.

### CODE REVIEW

File: rpcs3/Emu/NP/rpcn_client.cpp [hunk 1]
```diff
@@ -1590,11 +1590,14 @@ namespace rpcn
...
 	}
 
-	bool rpcn_client::req_ticket(u32 req_id, const std::string& service_id)
+	bool rpcn_client::req_ticket(u32 req_id, const std::string& service_id, const std::vector<u8>& cookie)
 	{
 		std::vector<u8> data;
 		std::copy(service_id.begin(), service_id.end(), std::back_inserter(data));
 		data.push_back(0);
+		const le_t<u32> size = cookie.size();
+		std::copy(&size, &size + sizeof(le_t<u32>), std::back_inserter(data));
```
Comment (elad335) [inline rpcs3/Emu/NP/rpcn_client.cpp:1599]: There must be a bug here, this isn't std::fill.

Comment (RipleyTom) [reply rpcs3/Emu/NP/rpcn_client.cpp]: better now? :)

Comment (elad335) [reply rpcs3/Emu/NP/rpcn_client.cpp]: I mean, I do not see how &size + sizeof(le_t<u32>) - 1 is a valid memory location to address.

Comment (RipleyTom) [reply rpcs3/Emu/NP/rpcn_client.cpp]: It's not but that's why I changed the code to
```cpp
std::copy(reinterpret_cast<const u8 *>(&size), reinterpret_cast<const u8 *>(&size) + sizeof(le_t<u32>), std::back_inserter(data));
```

Comment (elad335) [reply rpcs3/Emu/NP/rpcn_client.cpp]: Ah ok. 👍

**Pull request merged**